### PR TITLE
Add 3.0 section for What's new (on the site)

### PR DESCRIPTION
**Pull request by parlough**
I went with a bit of a different style since it was a major release and we did a major restructuring. It ended up a bit longer because of that, but I think it should be fine. Needs a good copy edit though _(thanks!_). I also added videos this time.

Closes https://github.com/dart-lang/site-www/issues/4830

**Staged:** https://dart-dev--pr5116-feature-whats-new-3-3qbscwhc.web.app/guides/whats-new

### CODE REVIEW

File: src/_guides/whats-new.md [hunk 1]
```diff
@@ -17,6 +17,117 @@ and follow the [Dart blog][].
...
+and the [SDK changelog][].
+
+[Announcing Dart 3]: https://medium.com/dartlang/announcing-dart-3-53f065a10635
+[SDK changelog]: https://github.com/dart-lang/sdk/blob/main/CHANGELOG.md#300---2023-05-10
+
+### Docs updated or added to dart.dev
+{: .no_toc}
+
+In preparation for Dart 3, we split up and reorganized the
+constantly growing language tour into new conceptual categories.
```
Comment (MaryaBelanger) [inline src/_guides/whats-new.md:35]:
```suggestion
Language Tour into new conceptual categories.
```

File: src/_guides/whats-new.md [hunk 2]
```diff
@@ -17,6 +17,117 @@ and follow the [Dart blog][].
...
+  better understand how they combine.
+
+To help you transition to Dart 3's enforcement of [sound null safety][]
+and other changes, we also prepared the following updates:
+
+- Created a comprehensive [Dart 3 migration guide][].
+- Migrated all documentation and code examples
+  to Dart 3, the latest tool releases, and the latest dependencies.
+- Clarified Dart's null-safe type system and 
+  how that always applies in Dart 3.
```
Comment (MaryaBelanger) [inline src/_guides/whats-new.md:60]:
```suggestion
- Clarified that type system is now always null-safe in Dart 3.
```

File: src/_guides/whats-new.md [hunk 3]
```diff
@@ -17,6 +17,117 @@ and follow the [Dart blog][].
...
+- Updated and reorganized the [language evolution][] page
+  and its [language versioning][] discussion.
```
Comment (MaryaBelanger) [inline src/_guides/whats-new.md:62]:
```suggestion
- Updated and reorganized the [Language evolution][] page
  and its [Language versioning][] discussion.
```

File: src/_guides/whats-new.md [hunk 4]
```diff
@@ -17,6 +17,117 @@ and follow the [Dart blog][].
...
+- Removed remnants of Dart 1 and early Dart 2 docs, notes, and resources.
+
+In addition to all of that Dart 3 goodness and general site improvements,
+we made the following changes:
```
Comment (MaryaBelanger) [inline src/_guides/whats-new.md:66]: Hahah "goodness" is a good one!
```suggestion
In addition to the new Dart 3 content and corresponding updates across the site,
we made the following changes:
```

**Review by MaryaBelanger — APPROVED** [review]
Great idea with the videos! Just a few small changes, thank you!

**Comment by parlough** [comment]
> Great idea with the videos! Just a few small changes, thank you!

Thank _you!_

**Pull request merged**